core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -21,27 +21,36 @@
 
 import com.github.benmanes.caffeine.cache.Caffeine;
 import com.github.benmanes.caffeine.cache.LoadingCache;
-import com.google.common.collect.ImmutableList;
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
```

Comment:
We prefer using `java.util.function` classes instead of Guava.

Reply:
I'm slightly conflicted here. 
1. We heavily tend to use `Iterables` which require `google.Predicate`
1. Predicates are java.util.Functions as they implement that interface
1. minor:  Predicates provide alwaysTrue() and alwaysFalse() which helps us to simplify our compound predicates.

What do u think?

Reply:
The main factor in not using Guava functions is that we don't want to leak Guava in the Iceberg API. That means where we accept functions, we should use the ones from `java.util.function`. From there, I don't think it is a good idea to mix two nearly identical but different interfaces in separate parts of the code.

It's not too bad to adapt the standard Java interfaces to Guava. Usually it's just a matter of passing `javaFunc::apply`.